cmd/kubeturbo/app/kubeturbo_builder.go

```diff
@@ -137,44 +117,35 @@ func (s *VMTServer) Run(_ []string) error {
 	// 	&clientcmd.ClientConfigLoadingRules{ExplicitPath: s.Kubeconfig},
 	// 	&clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: s.Master}}).ClientConfig()
 
-	kubeconfig, err := clientcmd.BuildConfigFromFlags(s.Master, s.Kubeconfig)
+	kubeConfig, err := clientcmd.BuildConfigFromFlags(s.Master, s.Kubeconfig)
 	if err != nil {
 		glog.Errorf("Error getting kubeconfig:  %s", err)
 		return err
 	}
 	// This specifies the number and the max number of query per second to the api server.
-	kubeconfig.QPS = 20.0
-	kubeconfig.Burst = 30
+	kubeConfig.QPS = 20.0
+	kubeConfig.Burst = 30
 
-	kubeClient, err := client.New(kubeconfig)
+	kubeClient, err := client.New(kubeConfig)
 	if err != nil {
 		glog.Fatalf("Invalid API configuration: %v", err)
 	}
 
-	leaderElectionClient, err := clientset.NewForConfig(restclient.AddUserAgent(kubeconfig, "leader-election"))
+	leaderElectionClient, err := clientset.NewForConfig(restclient.AddUserAgent(kubeConfig, "leader-election"))
 	if err != nil {
 		glog.Fatalf("Invalid API configuration: %v", err)
 	}
 
-	// serverAddr, targetType, nameOrAddress, targetIdentifier, password
-	var vmtMeta *metadata.VMTMeta
-	if s.TurboServerAddress != "" && s.OpsManagerUsername != "" && s.OpsManagerPassword != "" {
-		vmtMeta, err = metadata.NewVMTMeta(s.TurboServerAddress, s.TurboServerPort, s.OpsManagerUsername, s.OpsManagerPassword, kubeconfig.Host)
-	} else if s.MetaConfigPath != "" {
-		vmtMeta, err = metadata.NewVMTMetaFromFile(s.MetaConfigPath, kubeconfig.Host)
-	} else {
-		glog.Errorf("Failed to build turbo config.")
-		os.Exit(1)
-	}
+	glog.V(3).Infof("spec path is: %v", s.K8sTAPSpec)
+
+	k8sTAPSpec, err := kubeturbo.ParseK8sTAPServiceSpec(s.K8sTAPSpec)
 	if err != nil {
-		glog.Errorf("Get error when generating turbo config: %s", err)
+		glog.Errorf("Failed to generate correct TAP config: %s", err)
 		os.Exit(1)
 	}
 
-	glog.V(3).Infof("Finished creating turbo configuration: %++v", vmtMeta)
-
-	etcdclientBuilder := etcdhelper.NewEtcdClientBuilder().ServerList(s.EtcdServerList).SetTransport(s.EtcdCA, s.EtcdClientCertificate, s.EtcdClientKey)
-	etcdClient, err := etcdclientBuilder.CreateAndTest()
+	etcdClientBuilder := etcdhelper.NewEtcdClientBuilder().ServerList(s.EtcdServerList).SetTransport(s.EtcdCA, s.EtcdClientCertificate, s.EtcdClientKey)
+	etcdClient, err := etcdClientBuilder.CreateAndTest()
```

Comment:
This is to create the etcdClient and then test if the etcdClient can reach the etcd cluster.